### PR TITLE
feat(preview): Vuetify 3 + Vite preset

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
### To test
```bash
vue create my-app
```

#### Vue prompt
```bash
? Please pick a preset: (Use arrow keys)
    Default ([Vue 2] babel, eslint) 
  ❯ Default (Vue 3 Preview) ([Vue 3] babel, eslint) 
    Manually select features 
```

#### cd into app dir
```bash
cd my-app
```

#### Install vuetify/vue-cli-plugins
```bash
yarn add -D '../local/path/to/vue-cli-plugins-vuetify'
```

#### Invoke vuetify
```bash
vue invoke vuetify
```

#### Vuetify prompt
```bash
? Choose a preset: (Use arrow keys)
  Default (recommended) 
❯ Preview (v3 + vite)
  Prototype (rapid development) 
  V3 (alpha) 
  Configure (advanced) 
```

#### Run app
```bash
yarn dev
```

### CODE REVIEW

File: packages/vue-cli-plugin-vuetify/generator/index.js [hunk 1]
```diff
@@ -19,6 +21,28 @@ module.exports = (api, opts) => {
...
+  if (opts.useVite) {
+    api.extendPackage({
+      devDependencies: {
+        '@vitejs/plugin-vue': '^1.1.5',
+        'vite': '^2.0.5',
+      },
+      scripts: {
+      'serve': 'vite preview',
+      'build': 'vite build',
+      'dev': 'vite',
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/index.js:34]:
```suggestion
        'serve': 'vite preview',
        'build': 'vite build',
        'dev': 'vite',
```

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue [hunk 2]
```diff
@@ -12,7 +12,11 @@
 
       <v-col class="mb-4">
         <h1 class="display-2 font-weight-bold mb-3">
-          Welcome to Vuetify V3 Alpha
+          <%_ if (useVite) { _%>
+          Welcome to Vuetify 3 + Vite Preview
+          <%_ } else { _%>
+          Welcome to Vuetify 3 Alpha
+          <%_ } _%>
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue:19]:
```suggestion
          <div>Welcome to the Vuetify 3 Alpha</div>
```

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue [hunk 3]
```diff
@@ -12,7 +12,11 @@
...
         </h1>
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue:21]:
```suggestion

<%_ if (useVite) { _%>
          <small>Vite Preview</small>
<%_ } _%>

```

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/vite.config.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { defineConfig } from 'vite'
+import vue from '@vitejs/plugin-vue'
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [vue()],
+  define: {
+    'process.env': {},
+  },
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/vite.config.js:9]:
```suggestion
  define: { 'process.env': {} },
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Comment by Djaler** [comment]
How ironically, isn't it? Using old tooling to install the new one

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/index.vite.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8" />
+    <link rel="icon" href="/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite App</title>
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/index.vite.html:8]:
```suggestion
    <title>Vuetify 3 Vite Preview</title>
```

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue [hunk 6]
```diff
@@ -92,57 +96,59 @@
...
+    ],
+    logo,
+    whatsNext: [
+      {
+        text: 'Explore components',
+        href: 'https://vuetifyjs.com',
+      },
+      {
+        text: 'Roadmap',
+        href: 'https://vuetifyjs.com/en/introduction/roadmap/',
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue:145]:
```suggestion
        href: 'https://vuetifyjs.com/introduction/roadmap/',
```

File: packages/vue-cli-plugin-vuetify/util/presets.js [hunk 7]
```diff
@@ -4,8 +4,9 @@ const resolve = file => path.resolve(__dirname, file)
 
 const PRESET_MAP = {
   default: 'Default (recommended)',
+  preview: 'Preview (vuetify 3 + vite)',
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/util/presets.js:7]:
```suggestion
  preview: 'Preview (Vuetify 3 + Vite)',
```

**Review by johnleider — CHANGES_REQUESTED** [review]
Looking really good 😀

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue (hunk 6 above)
Comment (ElijahKotyluk) [reply packages/vue-cli-plugin-vuetify/generator/templates/v3/src/components/HelloWorld.js.vue:145]: Ah, good catch!

**Review by johnleider — APPROVED** [review]

**Comment by vallemar** [comment]
Hi @johnleider,

When is this functionality expected to be available?

Thanks!

**Comment by johnleider** [comment]
As soon as it's released.

**Pull request merged**